element-connector/src/main/java/org/eclipse/californium/elements/TlsCorrelationContext.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations GmbH - add support for correlation context to provide
+ *                                      additional information to application layer for
+ *                                      matching messages (fix GitHub issue #1)
+ ******************************************************************************/
+package org.eclipse.californium.elements;
+
+import org.eclipse.californium.elements.util.StringUtil;
+
+/**
+ * A correlation context that explicitly supports TLS specific properties.
+ * Currently not aware of renegotiation (API to acquire missing).
+ */
+public class TlsCorrelationContext extends TcpCorrelationContext {
+
+	public static final String KEY_SESSION_ID = "SESSION_ID";
```

Comment:
In TLS I can't access the "epoch" (at least, right now :-) ).
I think, having "special" classes for each type of connection, is useful. It makes it easier to distinguish.